parse_mem.py

```diff
@@ -1,151 +1,90 @@
-#!/usr/bin/python
-
 import sys
 import base64
+
 from pyasn1.type import univ
 from pyasn1.codec.der import encoder
 
+magic_headers = [
+    "ssh-rsa", "ssh-dss", "ecdsa-sha2-nisp256", "ssh-ed25519"
+]
 
-class sshkeyparse:
-    """ This class is designed to parse a memory dump of ssh-agent and create
-    unencrypted ssh keys that can then be used to gain access to other
-    systems"""
-    keytypes = {
-        'rsa': "ssh-rsa",
-        'dsa': "ssh-dss",
-        'ecsda': "ecdsa-sha2-nisp256",
-        'ed25519': "ssh-ed25519"
-    }
-
-    def read(self, memdump):
-        """ Reads a file and stories it in self.mem"""
-        self.inputfile = memdump
-        file = open(memdump, 'rb')
-        self.mem = "".join(file.readlines())
-        file.close()
-
-    def unpack_bigint(self, buf):
-        """Turn binary chunk into integer"""
-
-        v = 0
-        for c in buf:
-            v *= 256
-            v += ord(c)
 
-        return v
+class SshKeyParse:
+    """ 
+    This class is designed to parse a memory dump of ssh-agent and create
+    unencrypted ssh keys that can then be used to gain access to other
+    systems
+    """
+    def __init__(self):
+        self.file_path = None
+        self.file_content = None
+        
+    def read(self, fp):
+        """ 
+        Reads a file and stories it in self.file_content
+        """
+        self.file_path = fp
+        _f = open(fp, 'rb')
+        self.file_content = _f.read()
+        _f.close()
 
     def search_key(self):
-        """Searches for keys in self.mem"""
-
-        keysfound = {}
-
-        for type in self.keytypes:
-            magic = self.mem.find(self.keytypes[type])
-
-            if magic is not -1:
-                keysfound[magic] = type
-
-        if keysfound:
-            print ("Found %s key" % keysfound[sorted(keysfound)[0]])
-            self.mem = self.mem[sorted(keysfound)[0]:]
-            self.type = keysfound[sorted(keysfound)[0]]
-            return 1
-
-        if not keysfound:
-            return -1
-
-    def getkeys(self, output):
-        """ Parses for keys stored in ssh-agent's stack """
-
-        keynum = 0
-        validkey = 0
-
-        validkey = self.search_key()
-        while validkey != -1:
-
-            if keynum == 0:
-                keynum += 1
-                self.create_key(output)
-
+        """
+        Searches for keys in file content.
+        """
+        found_keys = {}
+        for magic in magic_headers:
+            offset = self.file_content.find(magic.encode(encoding='ascii'))
+            if offset:
+                print("Success find key type! key type:{}, key start offset:0x"
```

Comment:
Grammatical improvements needed here.

```suggestion
                print("Successfully found key type {} at offset:0x"
```